cpplang/parser.py

```diff
@@ -879,7 +879,7 @@ def parse_ParmVarDecl(self, node) -> tree.ParmVarDecl:
                                 default=default, attributes=attributes)
 
     def as_statement(self, subnode):
-        if isinstance(subnode, tree.Expression):
+        if isinstance(subnode, tree.Expression) or isinstance(subnode, tree.ExprWithCleanups):
```

Comment:
Why not having `ExprWithCleanups` derive from Expression instead?

Reply:
It would definitely make more sense to me but as it was not already the case I initially assumed it was intentional that it was not the case. My mistake, I'll go with inheritance, this line felt wrong indeed.